packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -516,7 +516,9 @@ export async function createRouteManifest(
 		...[...filteredFiledBasedRoutes, ...injectedRoutes, ...redirectRoutes].sort(routeComparator),
 	];
 
-	settings.buildOutput = getPrerenderDefault(config) ? 'static' : 'server';
+	// NOTE: `buildOutput` may already be set by `runHookConfigDone` if the adapter only
+	// support server build output
+	settings.buildOutput ??= getPrerenderDefault(config) ? 'static' : 'server';
```

Comment:
So the reordering of hooks is for this to work right? Just to make sure I understand, we don't want to infer the build output if the adapter locks it. Is that right?

Reply:
Yup. This code will only work as a fallback